### PR TITLE
SLING-10458 use NIO files

### PR DESCRIPTION
**Pull request by kwin**

### CODE REVIEW

File: src/main/java/org/apache/sling/feature/io/artifacts/spi/package-info.java [hunk 1]
```diff
@@ -17,7 +17,7 @@
  * under the License.
  */
 
-@org.osgi.annotation.versioning.Version("1.0.0")
+@org.osgi.annotation.versioning.Version("2.0.0")
```
Comment (kwin) [inline src/main/java/org/apache/sling/feature/io/artifacts/spi/package-info.java:20]: Is there a need to stay backwards compatible?

File: src/main/java/org/apache/sling/feature/io/artifacts/ArtifactHandler.java [hunk 2]
```diff
@@ -46,11 +47,24 @@ public ArtifactHandler(final String url, final URL localURL) {
      * @param file The file for the artifact
      * @throws MalformedURLException If the file name cannot be converted to a URL.
      * @since 1.1.0
+     * @deprecated Use {@link #ArtifactHandler(Path)} instead
      */
+    @Deprecated
```
Comment (cziegeler) [inline src/main/java/org/apache/sling/feature/io/artifacts/ArtifactHandler.java]: Why should we deprecate methods using File? Today java.io.File is not deprecated and used everywhere, so it seems convenient to have this

Comment (kwin) [reply src/main/java/org/apache/sling/feature/io/artifacts/ArtifactHandler.java]: I agree, fixed in https://github.com/apache/sling-org-apache-sling-feature/pull/26/commits/fafa5e407baeed8ec0f7113951f8c0c8f6d37d7f.

**Comment by cziegeler** [comment]
Lgtm, thanks. @karlpauls WDYT - is this still working with url handling for artifacts?

**Review by karlpauls — APPROVED** [review]
@cziegeler, hopefully it will. Devil is in the details I guess but as long as the toUri() and to URL() works the same it should be ok.

**Pull request merged**